src/main/java/uk/gov/hmcts/reform/hmc/api/mapper/AutomatedHearingTransactionRequestMapper.java

```diff
@@ -0,0 +1,575 @@
+            .listingComments(hearingData.getAdditionalHearingDetails())
+            .hearingRequester(hearingData.getHearingJudgePersonalCode())
+            .privateHearingRequiredFlag(C100.equals(caseData.getCaseTypeOfApplication()))
+            .panelRequirements(new PanelRequirements())
+            .leadJudgeContractType("")
+            .hearingIsLinkedFlag(hearingData.getHearingListedLinkedCases() != null)
+            .hearingChannels(List.of(hearingData.getHearingChannelsEnum().getDisplayedValue()))
+            .build();
+    }
+
+    private static HearingWindow hearingWindow(HearingData hearingData) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        if (hearingData.getEarliestHearingDate() != null
+            || hearingData.getLatestHearingDate() != null || hearingData.getFirstDateOfTheHearing() != null) {
+            return HearingWindow.hearingWindowWith()
+                .dateRangeStart(hearingData.getEarliestHearingDate() != null
+                                    ? hearingData.getEarliestHearingDate().format(formatter) : null)
+                .dateRangeEnd(hearingData.getLatestHearingDate() != null ? hearingData.getLatestHearingDate().format(
+                    formatter) : null)
+                .firstDateTimeMustBe(hearingData.getFirstDateOfTheHearing() == null ? null :
+                                         dateOfHearing(
+                                             hearingData.getFirstDateOfTheHearing().format(formatter),
+                                             hearingData.getHearingMustTakePlaceAtHour(),
+                                             hearingData.getHearingMustTakePlaceAtMinute()
+                                         ))
+                .build();
+        }
+        return null;
+    }
+
+    private static int hearingDuration(String days, String hours, String minutes) {
+
+        if (days != null) {
+            return Integer.parseInt(days) * 360;
+        } else if (hours != null) {
+            return Integer.parseInt(hours) * 60;
+        } else if (minutes != null) {
+            int remainder = Integer.parseInt(minutes) % 5;
+            if (remainder != 0) {
+                return (Integer.parseInt(minutes) / 5) * 5 + 5;
+            }
+        }
+        return 0;
+    }
+
+    private static String dateOfHearing(@NotNull String firstDate, String hours, String minutes) {
+        return String.format("%sT%s:%s:00Z", firstDate, hours != null ? hours : "00", minutes != null ? minutes : "00");
+    }
+
+    private static int noOfPhysicalAttendees(String attendSameWayYesOrNo, HearingData hearingData) {
+        int totalParticipants = 0;
+        if (YesOrNo.YES.name().equalsIgnoreCase(attendSameWayYesOrNo)
+            && hearingData.getHearingChannelsEnum() == HearingChannelsEnum.INTER) {
+            List<String> noOfParticipants = new ArrayList<>();
+            if (hearingData.getHearingDataApplicantDetails() != null) {
```

Comment:
Should we check case type of application? and add a null check

Reply:
hi @shashisk97 yes added the case type of application check as well ... Thank you ... completed all the review comments @shashisk97 @yogendraU Thank you !!